### ISSUE TITLE
Use GitHub Actions workflows to automate building/packaging SCT's binary dependencies

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#3788]
> **Note**: This issue is meant to describe the problems that @kousu's currently-stalled draft PRs (#2666, #2667, #2668) are trying to solve.

## Context

SCT relies on a few C++ dependencies that must be compiled manually. But, this process is typically quite tedious, complex, and and error prone when done on a personal computer. (See: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3682#issuecomment-1036485590 for an example of the process, as well as [this dev wiki page](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-How-to-compile-binaries-(ANTs,-isct)) for a less thorough overview). Using a personal computer also risks generating binaries that don't preserve the results that SCT's existing binaries produce.

Because of these problems, @kousu began an effort 2 years ago to use GitHub Actions to compile and package our binaries. If completed, this would save SCT developers the trouble of figuring out how to set up and successfully run a build environment. This would also allow us to more easily update our binaries to keep up with upstream changes (e.g. as was required by https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2642).

I've taken some steps to refresh this effort by fixing problems that have accumulated in the years since. (See [this PR](https://github.com/spinalcordtoolbox/spinalcordtoolbox-ants/pull/1).)  The remaining TODOs are outlined below:

## Remaining work

* Repo organization:
    * [x] Move kousu/ANTs to [spinalcordtoolbox/spinalcordtoolbox-ants](https://github.com/spinalcordtoolbox/spinalcordtoolbox-ants)
    * [x] Move kousu/spinalcordtoolbox-binaries to [spinalcordtoolbox/spinalcordtoolbox-binaries](https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries)

* https://github.com/spinalcordtoolbox/spinalcordtoolbox-ants/pull/1:
    - [x] Either debug `centos-stream8` and `archlinux`, or cut our losses and temporarily disable them (since they aren't explicitly used right now).
    - [x] Change `macos-latest` to either `macos-10.15` or `macos-11.0` (depending on whatever was current 2 years ago)
    - [x] Re-run the workflow to generate new binaries + make sure the `release` step functions okay + the macOS binaries match.
    - [x] Tweak the workflow to allow us to specify the build commit via `workflow_dispatch`.
    - [x] Merge the PR, then generate a release on master.

* [spinalcordtoolbox/spinalcordtoolbox-binaries](https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries)
    - [ ] Tweak the repo's [`build.yml`](https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries/blob/master/.github/workflows/build.yml) workflow to package up the licenses alongside the binaries (superseding [#2666](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2666))
    - [x] Tweak the repo's [`build.yml`](https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries/blob/master/.github/workflows/build.yml) workflow to fetch the latest `ctrDetect` binaries (superseding [#2667](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2667))
    - [x] Update the links in [#2668](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2668).
    - [ ] Update SCT's internal dev documentation to assist future devs in navigating binaries.

(NB: Optionally, we might also want to try automating Windows, too, since we now have really good clarity into what is needed to build binaries on Windows (see: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3682#issuecomment-1036485590).)

**Comment by joshuacwnewton** [comment]
> Note: This issue is meant to describe the problems that @kousu's currently-stalled draft PRs (https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2666, https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2667, https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2668) are trying to solve.

To provide a little more context for the 3 draft PRs:

* PRs [#2666](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2666), [#2667](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2667) and [#2668](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2668) are all trying to update the OSF binary links.
* But, since those PRs, a more recently merged PR [#2821](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2821) had _also_ changed the OSF links (https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2821/commits/18b0fe64ae3c7664dc6d521b7d5d3da5e87b791d), but with the purpose of updating the ANTs binaries instead!

So, there are actually _5_ sets of links:

1. `binaries_linux`
    * Old link (i.e. previous master): [`2020-04-20`](https://osf.io/mka78/?action=download)
    * Proposed link 1 (i.e. this PR): [`2020-04-21`](https://osf.io/admxy/?action=download)
    * Proposed link 2 (i.e. PR #[2667](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2667)): [`2020-04-22`](https://osf.io/xhr4v/?action=download)
    * Proposed link 3 (i.e. PR #[2668](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2668)): [`kousu/spinalcordtoolbox-binaries`](https://github.com/kousu/spinalcordtoolbox-binaries/releases/download/refs%2Fpull%2F3%2Fmerge-85120650/spinalcordtoolbox-binaries_linux.tar.gz)
    * Current link (i.e. present master): [`2020-08-01`](https://osf.io/cs6zt/?action=download)

2. `binaries_osx`
    * Old link (i.e. previous master): [`2020-04-20`](https://osf.io/dn67h/?action=download)
    * Proposed link 1 (i.e. this PR): [`2020-04-21`](https://osf.io/7vx5g/?action=download)
    * Proposed link 2 (i.e. PR #[2667](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2667)): [`2020-04-22`](https://osf.io/eg2m5/?action=download)
    * Proposed link 3 (i.e. PR  #[2668](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2668)): [`kousu/spinalcordtoolbox-binaries`](https://github.com/kousu/spinalcordtoolbox-binaries/releases/download/refs%2Fpull%2F3%2Fmerge-85120650/spinalcordtoolbox-binaries_osx.tar.gz)
    * Current link (i.e. present master): [`2020-08-01`](https://osf.io/874cy?action=download)

tl;dr I believe what's necessary is to create 1 final OSF package that combines the changes from #2666, #2667, #2668, and #2821.
